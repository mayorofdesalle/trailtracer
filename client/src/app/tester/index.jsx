import '@testing-library/jest-dom';

import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import './mocks/i18n.mock';
import './mocks/matchMedia.mock';
import './mocks/resizeObserver.mock';

import store from '@app/store';
import GlobalStyle from '@components/style/GlobalStyle';
import Theme from '@components/style/Theme';

const Providers = ({ children }) => {
    return (
        <Provider store={store}>
            <Theme>
                <GlobalStyle />
                <BrowserRouter>
                    {children}
                </BrowserRouter>
            </Theme>
        </Provider>
    );
};

const customRender = (ui, options) =>
    render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
export * from 'vitest';

// override render method
export { customRender as render };
