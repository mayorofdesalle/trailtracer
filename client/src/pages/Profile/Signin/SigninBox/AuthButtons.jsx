import { useContext } from 'react';
import styled from 'styled-components';

import Button from '@components/ui/Button';
import Container from '@components/ui/Container';
import Icon from '@components/ui/Icon';
import PageContext from '@context/PageContext';

const AuthButtonsInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${Button} {
        --size: clamp(2.5rem, max(3dvw, 3dvh), 5rem);
        width: var(--size);
        height: var(--size);

        & > svg {
            height: 100%;
            width: 100%;
        }
    }
`;

/**
 * AuthButtons
 * @description
 * Styled container that wraps the auth buttons.
 **/
const AuthButtons = () => {
    const theme = useContext(PageContext).theme;
    const animColors = [theme.colors.background, theme.colors.secondary];

    return (
        <AuthButtonsInner>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='google-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='facebook-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='twitter-fill' color={theme.colors.primary} />
            </Button>
        </AuthButtonsInner>
    );
};

export default AuthButtons;