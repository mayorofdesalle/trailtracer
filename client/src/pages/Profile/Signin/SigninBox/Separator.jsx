import { useContext } from 'react';
import styled from 'styled-components';

import Container from '@components/ui/Container';
import Text from '@components/ui/Text';
import PageContext from '@context/PageContext';

const HR = styled.hr`
    width: 100%;
    height: 2px;
    margin-left: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const SeparatorInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${HR}:first-of-type {
        margin-left: 0;
        margin-right: clamp(0.5rem, min(1dvw, 1dvh), 2rem);
    }

    & > ${Text} {
        flex-shrink: 0;
    }
`;

/**
 * Separator
 * @description
 * Horizontal rule with text in the middle.
 * **/
const Separator = () => {
    const context = useContext(PageContext);
    const theme = context.theme;
    const t = context.t;

    return (
        <SeparatorInner>
            <HR /><Text $color={theme.colors.background}>{t('signinPage.auths')}</Text><HR />
        </SeparatorInner>
    );
};

export default Separator;