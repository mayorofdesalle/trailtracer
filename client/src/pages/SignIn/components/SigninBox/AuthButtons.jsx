import styled, { useTheme } from 'styled-components';

import Container from '@components/ui/Container';
import Button from '@components/ui/Button';
import Icon from '@components/ui/Icon';

const AuthButtonsInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;
`;

const AuthButtons = () => {
    const theme = useTheme();
    const animColors = [theme.colors.background, theme.colors.secondary];

    return (
        <AuthButtonsInner>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='google-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='facebook-fill' color={theme.colors.primary} />
            </Button>
            <Button $bgColor={theme.colors.background} $animColors={animColors}>
                <Icon name='twitter-fill' color={theme.colors.primary} />
            </Button>
        </AuthButtonsInner>
    );
};

export default AuthButtons;