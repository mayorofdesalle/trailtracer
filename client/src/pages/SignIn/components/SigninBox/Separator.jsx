import styled, { useTheme } from 'styled-components';

import Container from '@components/ui/Container';
import Text from '@components/ui/Text';

const HR = styled.hr`
    width: calc((100% - clamp(0.825rem, min(2dvw, 2dvh), 2.5625rem) * 9) / 2);
    height: 2px;
    background-color: ${({ theme }) => theme.colors.secondary};
`;

const SeparatorInner = styled(Container)`
    justify-content: space-around;
    height: fit-content;

    & > ${HR} {
        margin-left: 0.5rem;
    }

    & > ${HR}:first-of-type {
        margin-left: 0;
        margin-right: 0.5rem;
    }
`;

const Separator = () => {
    const theme = useTheme();
    return (
        <SeparatorInner>
            <HR /><Text $color={theme.colors.background}>or continue with</Text><HR />
        </SeparatorInner>
    );
};

export default Separator;