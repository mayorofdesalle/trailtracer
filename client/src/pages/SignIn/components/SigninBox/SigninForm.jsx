import styled, { useTheme } from 'styled-components';

import Input from '@components/ui/Input';
import Button from '@components/ui/Button';
import Text from '@components/ui/Text';
import Container from '@components/ui/Container';

const FormInner = styled.form`
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    height: fit-content;
    
    & > * {
        margin-top: clamp(1.5625rem, min(3dvw, 3dvh), 3.25rem);
    }

    & > ${Container}:first-of-type {
        margin-top: 0;
    }

    & > ${Button} {
        padding: 0.25rem;
        width: 100%;
    }
`;

const SigninForm = () => {
    const theme = useTheme();

    return (
        <FormInner>
            <>
                <label hidden>Username or email</label>
                <Input type='email' name='email' autoComplete='email' placeholder='Username or email' icon='at-fill' iconColor={theme.colors.primary} required />
            </>
            <>
                <label hidden>Password</label>
                <Input type='password' eyeColor={theme.colors.secondary} name='password' autoComplete='current-password' placeholder='Password' icon='key-fill' iconColor={theme.colors.primary} required />
            </>
            <Button $bgColor={theme.colors.background} $animColors={[theme.colors.background, theme.colors.secondary]}>
                <Text $color={theme.colors.primary} $heading>SIGN IN</Text>
            </Button>
        </FormInner>
    );
};

export default SigninForm;