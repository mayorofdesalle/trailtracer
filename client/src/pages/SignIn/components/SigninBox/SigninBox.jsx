import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import BentoBox from '@components/ui/Bento/BentoBox';
import Text from '@components/ui/Text';
import SigninForm from './SigninForm';
import Separator from './Separator';
import AuthButtons from './AuthButtons';

const SigninBoxInner = styled(BentoBox)`
    grid-area: SigninBox;
    flex-direction: column;
    justify-content: space-around;

    & > * {
        width: 80%;
    }
`;

const SigninBox = () => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <SigninBoxInner $color={theme.colors.primary}>
            <Text $color={theme.colors.background} $heading $ratio={1.5}>{t('signinPage.title')}</Text>
            <Text $color={theme.colors.background}>{t('signinPage.credentials')}</Text>
            <SigninForm />
            <Separator />
            <AuthButtons />
        </SigninBoxInner>
    );
};

export default SigninBox;