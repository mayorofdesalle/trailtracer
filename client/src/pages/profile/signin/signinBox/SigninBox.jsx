import { useTranslation } from 'react-i18next';
import styled, { useTheme } from 'styled-components';

import { Container, Text } from '@components/ui';
import { BentoBox } from '@components/ui/bento';

import Separator from '@pages/Profile//components/Separator';
import AuthButtons from '@pages/Profile/components/AuthButtons';
import SigninForm from './SigninForm';

const SigninBoxInner = styled(BentoBox)`
    grid-area: SignBox;
    flex-direction: column;
    justify-content: space-around;

    & > * {
        width: 80%;
    }

    & > ${Text} {
        width: 100%;
    }
`;

const Wrapper = styled(Container)`
    height: fit-content;
    flex-direction: column;
`;

/**
 * SigninBox
 * @description
 * Styled bento box to display the signin form and auth buttons.
 * **/
const SigninBox = () => {
    const theme = useTheme();
    const { t } = useTranslation();

    return (
        <SigninBoxInner $color={theme.colors.primary}>
            <Wrapper>
                <Text $color={theme.colors.background} $heading $ratio={1.5}>{t('signinPage.title')}</Text>
                <Text $color={theme.colors.background}>{t('signinPage.credentials')}</Text>
            </Wrapper>
            <SigninForm />
            <Separator color={theme.colors.background} />
            <AuthButtons bgColor={theme.colors.background} color={theme.colors.primary} animColors={[theme.colors.background, theme.colors.secondary]} />
        </SigninBoxInner>
    );
};

export default SigninBox;