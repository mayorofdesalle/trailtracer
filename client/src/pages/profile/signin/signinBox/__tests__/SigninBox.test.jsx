import { describe, expect, it, render, screen } from '@app/tester';

import SigninBox from '../SigninBox';

describe('SigninBox', () => {

    it('should render correctly', () => {
        render(<SigninBox />);

        const email = screen.getByRole('textbox');
        expect(email).toBeInTheDocument();

        const password = screen.getByLabelText(/password/i);
        expect(password).toBeInTheDocument();

        const buttons = screen.getAllByRole('button');
        expect(buttons).toHaveLength(4);
        buttons.forEach(button => expect(button).toBeInTheDocument());
    });

});