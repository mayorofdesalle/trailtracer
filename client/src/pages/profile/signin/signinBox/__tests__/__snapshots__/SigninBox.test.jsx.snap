// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SigninBox > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-guDLey sc-dmyCSP sc-blmEgr kHwwsn gVwHLx fIEBkl"
  >
    <div
      class="sc-guDLey sc-brSamD kHwwsn fJKiWp"
    >
      <div
        class="sc-gLLuof kZkuLk"
      >
        signinPage.title
      </div>
      <div
        class="sc-gLLuof gHTNhf"
      >
        signinPage.credentials
      </div>
    </div>
    <form
      class="sc-fHejqy cMecfl"
      role="form"
    >
      <div
        class="sc-guDLey sc-hABBmJ kHwwsn jomkDL"
      >
        <label
          for="email"
          hidden=""
        >
          forms.email
        </label>
        <input
          autocomplete="email"
          class="sc-ktwOfi lbIkZd"
          data-tooltip-id="email"
          id="email"
          name="email"
          placeholder="forms.email"
          type="email"
        />
        <svg
          class="sc-eDLKkx fuXsHX"
          fill="#435049"
          role="img"
        >
          <title>
            at-fill
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#at-fill"
          />
        </svg>
      </div>
      <div
        class="sc-guDLey sc-hABBmJ kHwwsn jomkDL"
      >
        <label
          for="password"
          hidden=""
        >
          forms.current-password
        </label>
        <input
          autocomplete="current-password"
          class="sc-ktwOfi kXHIUs"
          data-tooltip-id="password"
          id="password"
          name="current-password"
          placeholder="forms.current-password"
          type="password"
        />
        <svg
          class="sc-eDLKkx fuXsHX"
          fill="#435049"
          role="img"
        >
          <title>
            key-fill
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#key-fill"
          />
        </svg>
      </div>
      <button
        class="sc-beySPh jLMfrb"
        type="submit"
      >
        <div
          class="sc-gLLuof kEDizj"
        >
          signinPage.signin
        </div>
      </button>
    </form>
    <div
      class="sc-guDLey sc-csKJxZ kHwwsn dyIUBv"
      role="separator"
    >
      <hr
        class="sc-irLvIq hHRSQk"
        role="none"
      />
      <div
        class="sc-gLLuof gHTNhf"
      >
        signinPage.auths
      </div>
      <hr
        class="sc-irLvIq hHRSQk"
        role="none"
      />
    </div>
    <div
      class="sc-guDLey sc-eTNRI kHwwsn bwweEz"
    >
      <button
        class="sc-beySPh nsmIf"
        type="button"
      >
        <svg
          class="sc-eDLKkx fuXsHX"
          fill="currentColor"
          role="img"
        >
          <title>
            google-fill
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#google-fill"
          />
        </svg>
      </button>
      <button
        class="sc-beySPh nsmIf"
        type="button"
      >
        <svg
          class="sc-eDLKkx fuXsHX"
          fill="currentColor"
          role="img"
        >
          <title>
            facebook-fill
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#facebook-fill"
          />
        </svg>
      </button>
      <button
        class="sc-beySPh nsmIf"
        type="button"
      >
        <svg
          class="sc-eDLKkx fuXsHX"
          fill="currentColor"
          role="img"
        >
          <title>
            twitter-fill
          </title>
          <use
            href="http://localhost:3000/src/assets/icons.svg#twitter-fill"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
